apps/mobile/src/components/Markdown.tsx

```diff
@@ -103,14 +110,27 @@ export function Markdown({
 
   const handleLinkPress = useCallback(
     (url: string) => {
-      if (url && onBypassLinkPress) {
+      const isInternalLink = url.startsWith('https://gallery.so/');
+      const parsedUrl = new URL(url);
+      const splitBySlash = parsedUrl.pathname.split('/').filter(Boolean);
+
+      const isInternalLinkWithDeepLink =
+        isInternalLink &&
+        !(
+          typeof splitBySlash[0] === 'string' && KNOWN_NON_DEEPLINK_ROUTES.includes(splitBySlash[0])
+        );
```

Comment:
i think this check for `isInternalLinkWithDeepLink` could also be an external function exported out of `DeepLinkRegistrar`